### DISCUSSION TITLE
ARM support roadmap

### DISCUSSION THREAD

**Discussion by alexhaydock** [microsoft/SysmonForLinux#133] [General]
After a few attempts trying, and failing, to get Sysmon to build on an `aarch64` host, I'm assuming that this isn't supported due to the various bits of x86 ASM in the codebase.

Are there any plans within the roadmap for this project going forward to offer `aarch64` support at any point? I understand this could require rearchitecting quite a few components so might be a long-term-future delivery goal, but I'm wondering if it's a goal that's been considered at all yet at this point in time?

Thanks!

**Comment by Jiahao0** [comment]
Indeed, I believe this feature should be a big one. 
I'm searching for projects that support ARM, and work relatively well. At the end there aren't many. 
Actually, Sysmon was the first project I tried, but I could not get it running. Questions posted here are not getting response. I wonder if the project is being maintained any more. It's a bit frustrated to be honest. I really look forward to it working properly on ARM.

BTW, the thing is there are already a lot of decent eBPF applications, but almost all on x86, espacially for servers scenarios with docker.  Yet for ARM, I only got Tetragon working for the moment.

**Comment by MarioHewardt** [comment]
@alexhaydock   Sorry for the delayed response. Today, you are correct, we do not support ARM though it is on the longer-term roadmap (no committed date). 

@Jiahao0 The Sysinternals for Linux tools (which includes Sysmon for Linux) are very much being maintained and enhanced. ARM in particular is something that has been requested before and is on our longer-term roadmap. If there are specific questions that you feel haven't been answered I do apologize, and I'd love to get you some answers if you can point me to them. 

Thanks for the feedback, very much appreciated.
